stack.py

```diff
@@ -0,0 +1,29 @@
+class StackItem(object):
+    def __init__(self, data, next_item=None):
+        self.data = data
+        self.next_item = next_item
+
+    def __str__(self):
+        return self.data
+
+
+class StackFrame(object):
```

Comment:
In programming language design, the term `StackFrame` has a specific meaning.  This is not compatible with that meaning.  In fact, your `StackItem` class is closer in terms of purpose to the actual `Stack Frame` idea.  This class should simply be called a `Stack`